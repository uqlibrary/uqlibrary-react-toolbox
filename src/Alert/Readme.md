# Alert

Prorotype component to display alerts/warnings/errors

## Props
- `message: PropTypes.string`
- `title: PropTypes.string`
- `type PropTypes.string - valid values ['error', 'error_outline', 'warning', 'info', 'info_outline', 'help', 'help_outline']`
- `outsideLayout PropTypes.bool` - prop is to render the module to fit outside of .card-layout


## Usage
```jsx
import React from 'react';
...

import {Alerts} from 'uqlibrary-react-toolbox';
      
const App = () => (
  <div>
  
    <Alerts 
        title="This is a title"
        message="Some body text."
        type="error_outline"
        outsidelayout
    />
    
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

