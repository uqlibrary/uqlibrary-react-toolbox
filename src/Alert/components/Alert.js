import React from 'react';
import {PropTypes} from 'prop-types';
import FontIcon from 'material-ui/FontIcon';
import FlatButton from 'material-ui/FlatButton';
import IconButton from 'material-ui/IconButton';
import NavigationClose from 'material-ui/svg-icons/navigation/close';

const Alert = ({title, message, type, action, actionButtonLabel, allowDismiss, dismissAction}) => {
    return (
        <div className="forAlerts">
            <div className={type + ' alertWrapper'}>
                <div className="columns is-multiline is-mobile">
                    <div className="column is-narrow alertIcon">
                        <FontIcon className="material-icons">{type}</FontIcon>
                    </div>
                    <div className="column alertText">
                        <div><b>{title}</b>&nbsp;-&nbsp;{message}</div>
                    </div>
                    {
                        action && actionButtonLabel &&
                        <div className="column is-narrow is-12-mobile">
                            <FlatButton
                                label={actionButtonLabel}
                                onTouchTap={action}
                                className="alertAction"/>
                        </div>
                    }
                    {
                        allowDismiss && dismissAction &&
                        <div className="column is-narrow is-hidden-mobile">
                            <IconButton onTouchTap={dismissAction}>
                                <NavigationClose className="alertDismiss"/>
                            </IconButton>
                        </div>
                    }
                </div>
            </div>
        </div>
    );
};

Alert.propTypes = {
    message: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    type: PropTypes.oneOf(['error', 'error_outline', 'warning', 'info', 'info_outline', 'help', 'help_outline', 'done']),
    action: PropTypes.func,
    actionButtonLabel: PropTypes.string,
    allowDismiss: PropTypes.bool,
    dismissAction: PropTypes.func,
};

Alert.defaultProps = {
    message: 'Unexpected error',
    title: 'Error',
    type: 'error',
    allowDismiss: false
};

export default Alert;
