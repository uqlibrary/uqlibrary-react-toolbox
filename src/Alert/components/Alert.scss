@import '../../../sass/variables';

// Desktop
@media (min-width: 768px) {
  .alertWrapper {
    border-radius: 3px;
    margin: 24px 0 !important;
    padding: 12px;
    min-height: 40px;

    .alertIcon {
      float: left;
      margin-right: -6px;
      height: 50px;
      span {
        font-size: 50px !important;
      }
    }
    .alertText {
      font-weight: bold;
      min-height: 50px !important;
      margin: 0px !important;
      padding: 12px !important;
      .alertTitle {
        font-weight: bold !important;
      }
    }
  }

  // Help
  &.help, &.help_outline {
    border: 1px solid lighten($theme-color, 10%);
    background-color: darken($theme-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Error
  &.error, &.error_outline {
    border: 1px solid lighten($error-color, 10%);
    background-color: darken($error-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Warning
  &.warning {
    border: 1px solid lighten($warning-color, 10%);
    background-color: darken($warning-color, 0%);
    .alertText, .alertTitle {
      color: $text-color-reverse !important;
    }
    .alertIcon span {
      color: $text-color-reverse !important;
    }
  }

  // Info
  &.info, &.info_outline {
    border: 1px solid lighten($info-color, 10%);
    background-color: darken($info-color, 5%);
    .alertText, .alertTitle {
      color: $text-color-reverse;
    }
    .alertIcon span {
      color: rgba($text-color-reverse, 0.9) !important;
    }
  }

}

// Mobile
@media (max-width: 768px) {
  .alertWrapper {
    border-radius: 3px;
    margin: 24px 0 !important;
    padding: 0 12px 24px 12px;
    min-height: 40px;

    .alertIcon {
      float: left;
      margin-bottom: -20px;
      padding: 2x 12px 12px 6px !important;
      span {
        font-size: 30px !important;
        line-height: 40px;
      }
    }
    .alertText {
      font-weight: bold;
    }

    // Help
    &.help, &.help_outline {
      border: 1px solid lighten($theme-color, 10%);
      background-color: darken($theme-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Error
    &.error, &.error_outline {
      border: 1px solid lighten($error-color, 10%);
      background-color: darken($error-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Warning
    &.warning {
      border: 1px solid lighten($warning-color, 10%);
      background-color: darken($warning-color, 0%);
      .alertText, .alertTitle {
        color: $text-color-reverse !important;
      }
      .alertIcon span {
        color: $text-color-reverse !important;
      }
    }

    // Info
    &.info, &.info_outline {
      border: 1px solid lighten($info-color, 10%);
      background-color: darken($info-color, 5%);
      .alertText, .alertTitle {
        color: $text-color-reverse;
      }
      .alertIcon span {
        color: rgba($text-color-reverse, 0.9) !important;
      }
    }

  }
}