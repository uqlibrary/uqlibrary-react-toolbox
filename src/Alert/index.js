export { default as Alert } from './components/Alert';
