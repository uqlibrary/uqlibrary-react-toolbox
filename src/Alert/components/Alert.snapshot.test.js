jest.dontMock('../components/Alert');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Alert from '../components/Alert';

function setup({title, message, type}) {
    const props = {
        title: 'This is a title',
        message: 'This is a message',
        type: 'error'
    };
    return shallow(<Alert {...props} />);
}

describe('Alert snapshots test', () => {
    it('renders as expected', () => {
        const title = "This is a title";
        const message = "This is the message";
        const type = "error";

        const wrapper = setup(title, message, type);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});