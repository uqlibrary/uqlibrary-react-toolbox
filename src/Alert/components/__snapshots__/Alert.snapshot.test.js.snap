// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert snapshots test renders Alert of error type 1`] = `
<div
  className="error alertWrapper "
>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-narrow alertIcon"
    >
      <AlertError
        className="material-icons"
      />
    </div>
    <div
      className="column alertText"
    >
      <div>
        <b>
          This is a title - 
        </b>
        This is the message
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert snapshots test renders Alert of error type should render action button 1`] = `
<div
  className="error alertWrapper "
>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-narrow alertIcon linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <AlertError
        className="material-icons"
      />
    </div>
    <div
      className="column alertText linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <div>
        <b />
        Unexpected error
      </div>
    </div>
    <div
      className="column is-narrow-tablet is-12-mobile noDismiss"
    >
      <FlatButton
        className="alertAction"
        disabled={false}
        fullWidth={true}
        label="Do something"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[MockFunction]}
        primary={false}
        secondary={false}
      />
    </div>
  </div>
</div>
`;

exports[`Alert snapshots test renders Alert of error type should render dismiss icon button 1`] = `
<div
  className="error alertWrapper "
>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-narrow alertIcon"
    >
      <AlertError
        className="material-icons"
      />
    </div>
    <div
      className="column alertText"
    >
      <div>
        <b />
        Unexpected error
      </div>
    </div>
    <div
      className="column is-narrow-tablet is-hidden-mobile"
    >
      <IconButton
        className="alertDismissButton"
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        onTouchTap={[MockFunction]}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationClose
          className="alertDismiss"
        />
      </IconButton>
    </div>
  </div>
</div>
`;

exports[`Alert snapshots test renders Alert of error type should render dismiss icon button and action button 1`] = `
<div
  className="error alertWrapper "
>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-narrow alertIcon linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <AlertError
        className="material-icons"
      />
    </div>
    <div
      className="column alertText linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <div>
        <b />
        Unexpected error
      </div>
    </div>
    <div
      className="column is-narrow-tablet is-12-mobile"
    >
      <FlatButton
        className="alertAction"
        disabled={false}
        fullWidth={true}
        label="Do something"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[MockFunction]}
        primary={false}
        secondary={false}
      />
    </div>
    <div
      className="column is-narrow-tablet is-hidden-mobile"
    >
      <IconButton
        className="alertDismissButton"
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        onTouchTap={[MockFunction]}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationClose
          className="alertDismiss"
        />
      </IconButton>
    </div>
  </div>
</div>
`;

exports[`Alert snapshots test renders Alert of error type should render dismiss icon button and action button and spinner 1`] = `
<div
  className="error alertWrapper "
>
  <div
    className="columns is-multiline is-mobile"
  >
    <div
      className="column is-narrow alertIcon linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <CircularProgress
        className="alertSpinner"
        max={100}
        min={0}
        mode="indeterminate"
        size={32}
        thickness={4}
        value={0}
      />
    </div>
    <div
      className="column alertText linked"
      onClick={[MockFunction]}
      onKeyDown={[MockFunction]}
    >
      <div>
        <b />
        Unexpected error
      </div>
    </div>
    <div
      className="column is-narrow-tablet is-12-mobile"
    >
      <FlatButton
        className="alertAction"
        disabled={false}
        fullWidth={true}
        label="Do something"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[MockFunction]}
        primary={false}
        secondary={false}
      />
    </div>
    <div
      className="column is-narrow-tablet is-hidden-mobile"
    >
      <IconButton
        className="alertDismissButton"
        disableTouchRipple={false}
        disabled={false}
        iconStyle={Object {}}
        onTouchTap={[MockFunction]}
        tooltipPosition="bottom-center"
        touch={false}
      >
        <NavigationClose
          className="alertDismiss"
        />
      </IconButton>
    </div>
  </div>
</div>
`;
